src/provider/flagsmith/flagsmith_provider.py

```diff
@@ -0,0 +1,43 @@
+from numbers import Number
+from os import environ
+
+import flagsmith as Flagsmith
+
+from src.provider.provider import AbstractProvider
+
+
+class FlagsmithProvider(AbstractProvider):
+    def __init__(self, flagsmith: Flagsmith = None):
+        if flagsmith is None:
+            self.flagsmith_provider = Flagsmith(
+                environment_id=environ.get("FLAGSMITH_ENVIRONMENT_KEY", None)
+            )
+        else:
+            self.flagsmith_provider = flagsmith
+
+    def get_boolean_value(self, key: str, defaultValue: bool) -> bool:
+        value = self.flagsmith_provider.get_value(key)
+        if not isinstance(value, bool):
+            raise Exception()
```

Comment:
Does the spec have a defined behaviour for exception management anywhere? For example, should we have our own exceptions defined in `provider.flagsmith.exceptions` - maybe these should inherit from some base exception in the client?

Reply:
Also, interestingly, we have the concept of enabled and a boolean value. I think @dabeeeenster raised this but I'm not sure what the outcome was?

Reply:
The spec says no Exceptions should be raised so we should come up with a solution for what needs to be returned.

Reply:
@ajhelsby Correct, exceptions should never bubble up, the only possible situation where that's not the case is exceptions thrown in user-defined-code in error() and finally() hook stages, which is something we're hoping to clarify a bit more soon.

@matthewelwell The "enabled" state in Flagsmith is a bit of an outlier relative to most providers. We struggled with it for a while and decided not to reflect an equivalent concept in the spec.

It will be up to the FlagSmith provider implementation how to handle `enabled` vs getting a boolean value. It might be something that can be controlled/specified with an option to the FlagSmith provider.

Reply:
In the java side of things, we allow providers to throw exceptions, but they are caught by the client itself.

Reply:
>In the java side of things, we allow providers to throw exceptions, but they are caught by the client itself.

Great point, this is how the TS/JS SDK works as well.

Reply:
Regarding exceptions, are we ok with creating a base Exception class in the openfeature SDK and then expecting that all providers to subclass that or do we need to catch all Exceptions in the client. My preference would definitely be the former but it puts a responsibility on the providers to implement their exceptions correctly.

Reply:
We need to catch all exceptions in the client. No one is going to implement anything correctly ;) </snark>. The primary design principal I've been using is: "We should never throw for the user if we can avoid it, and certainly not surprisingly at runtime".  Like, I want people to be able to write code like:

```python
if client.get_boolean('show_redesign', False):
  return redesign()
return design()
```

and not this:

```python
try:
  if client.get_boolean('show_redesign', False):
    return redesign()
  return design()
except:
  return design()
```

They'd have to do the second if they want to catch stuff like connection timeouts, etc. We should assume an imperfect, but well intentioned provider.